### PR TITLE
Make sure that supported cipher suite is used, but not at the expense of others

### PR DESCRIPTION
**Pull request by j-baker**
HTTP/2 demands that "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256" is
supported. Dropwizard provides for this by _only_ letting users utilise
that cipher suite.

However, there are many other HTTP/2 supported cipher suites, which
at present Dropwizard does not let people use.

Here, we simply make sure that cipher exists in the supported list.

### CODE REVIEW

File: dropwizard-http2/src/main/java/io/dropwizard/http2/Http2ConnectorFactory.java [hunk 1]
```diff
@@ -123,4 +130,10 @@ public Connector build(Server server, MetricRegistry metrics, String name, Threa
                 new Jetty93InstrumentedConnectionFactory(sslConnectionFactory, metrics.timer(httpConnections())),
                 alpn, http2, http1);
     }
+
+    private static List<String> addIfNotPresent(List<String> someList, String elementToAddIfNotPresent) {
```
Comment (arteam) [inline dropwizard-http2/src/main/java/io/dropwizard/http2/Http2ConnectorFactory.java]: I would rename `someList` to `sourceList` for clarity and `elementToAddIfNotPresent` to `addedElement` for terseness.

**Comment by arteam** [comment]
Something like:
```java
SslContextFactory sslContextFactory = new SslContextFactory();
sslContextFactory.setTrustStorePath(ResourceHelpers.resourceFilePath("stores/http2_client.jts"));
sslContextFactory.setTrustStorePassword("http2_client");
sslContextFactory.setIncludeCipherSuites("TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384");

client = new HttpClient(new HttpClientTransportOverHTTP2(new HTTP2Client()), sslContextFactory);
client.start();

assertResponse(client.GET("https://localhost:" + appRule.getLocalPort() + "/api/test"));
```

**Comment by nickbabcock** [comment]
Hey @j-baker, did you get a chance take a look at arteam's suggestions? Sound reasonable?

**Comment by j-baker** [comment]
Ah, sorry. Yes, it's very reasonable, but I couldn't get it to work. I'll hopefully take another look soon.

**Comment by arteam** [comment]
Superseded by #2119

**Pull request closed without merging**